### ISSUE TITLE
Cannot add edition identifiers with names containing illegal Javascript id characters

### ISSUE THREAD

**Issue by tfmorris** [internetarchive/openlibrary#7032]
Selecting an book identifier which has a name/label containing parentheses (e.g. "Bibliothèque nationale de France (BnF)") causes a Javascript error when you attempt to add the identifier value. This causes one of the problems reported in #7030.

### Evidence / Screenshot (if possible)
![Screen Shot 2022-09-30 at 1 43 59 PM](https://user-images.githubusercontent.com/82178/193327113-467892ca-1dfb-4005-88a0-aa4537b0dfd9.png)

Uncaught Error: Syntax error, unrecognized expression: option[value=bibliothèque_nationale_de_france_(bnf)]
    at at.error (all.js?v=98dfd5206e072a58ac5eee040d2d626a:3:279628)
    at at.tokenize (all.js?v=98dfd5206e072a58ac5eee040d2d626a:3:287659)
    at at.select (all.js?v=98dfd5206e072a58ac5eee040d2d626a:3:288494)
    at Function.at [as find] (all.js?v=98dfd5206e072a58ac5eee040d2d626a:3:273083)
    at E.fn.init.find (all.js?v=98dfd5206e072a58ac5eee040d2d626a:3:291080)
    at eval (eval at u (all.js?v=98dfd5206e072a58ac5eee040d2d626a:3:379744), <anonymous>:6:35)
    at r (all.js?v=98dfd5206e072a58ac5eee040d2d626a:3:379807)
    at HTMLButtonElement.d (all.js?v=98dfd5206e072a58ac5eee040d2d626a:3:400061)
    at HTMLDocument.dispatch (all.js?v=98dfd5206e072a58ac5eee040d2d626a:3:309133)
    at m.handle (all.js?v=98dfd5206e072a58ac5eee040d2d626a:3:307128)

The string bibliothèque_nationale_de_france_(bnf) is not a legal Javascript identifier.

### Relevant url?
(https://openlibrary.org/books/OL32419698M/Hopeless/edit)

### Steps to Reproduce

1. Go to ...
2. Do ...

* Actual:
* Expected:

### Details

- **Logged in (Y/N)?**
- **Browser type/version?**
- **Operating system?**
- **Environment (prod/dev/local)?** prod

**Comment by tfmorris** [comment]
Labels containing dots/periods ("."), such as "abebooks.de" also fail, as described in #6300.

**Comment by JaydenTeoh** [comment]
@tfmorris Hi, I am trying to reproduce this bug but it seems to be resolved already (I am able to add an identifier containing parenthesis)
<img width="550" alt="image" src="https://user-images.githubusercontent.com/90945854/221371594-957c3888-bd48-482d-9dab-16d5d943955f.png">

**Comment by tfmorris** [comment]
Hi @JaydenTeoh. Thanks for working on this! The problem was not with the label, but with the "name" (ie ID). Can you confirm that you tested that?

The current config/edition.yml entry looks like:
```
-   label: Bibliothèque Nationale de France (BnF)
    name: bibliothèque_nationale_de_france
    notes: ''
    url: https://catalogue.bnf.fr/ark:/12148/@@@
```
which works, but the original failing version was 

```
-   label: Bibliothèque Nationale de France (BnF)
    name: bibliothèque_nationale_de_france_(bnf)
    notes: ''
    url: https://catalogue.bnf.fr/ark:/12148/@@@
```

It was the use of the value for `name` as an identifier in Javascript without quoting (and with embedded parentheses) which was causing the issue.

It looks like @RayBB and @scottbarnes have done some refactoring in this area, so it's possible they fixed the bug as a side effect of that, but please try the test case above.

**Comment by JaydenTeoh** [comment]
@tfmorris Hi, I can confirm that I am able to add edition identifiers using names with parenthesis. However, trying to add identifiers with "." and "," in the names are not successful (as mentioned in #6300). I'll look further into this issue and try to make a fix. Do let me know if there is anything else you would need me to test for this issue!

**Comment by JaydenTeoh** [comment]
@tfmorris I have opened a PR to fix this issue (https://github.com/internetarchive/openlibrary/pull/7575). Turns out all you gotta do is add some additional quotes! I am able to add identifiers/classifications with characters such as brackets, dots and commas now. Btw, may I ask how do I replicate production behavior on test environment? For example, I wanted to test the `Bibliothèque Nationale de France (BnF)` id but there is no such id on test environment.
